leap/rk/__init__.py

```diff
@@ -795,4 +801,157 @@ def generate(self):
 # }}}
 
 
+# {{{ Explicit SSP Runge-Kutta methods
+
+class SSPRKMethodBuilder(MethodBuilder):
+    r"""Explicit Strong Stability Preserving (SSP) Runge-Kutta Methods.
+
+    The methods are given in the now-standard Shu-Oscher form
+
+        .. math::
+
+            \begin{aligned}
+            y^{(i)} =\,\, & \sum_{k = 0}^{i - 1}
+                \alpha_{ik} y^{(i)} + \Delta t \beta_{ik} f(y^{(i)}), \\
+            y^{n + 1} =\,\, & y^{(n)},
+            \end{aligned}
+
+    for :math:`i \in \{1, \dots, n\}` and :math:`y^{(0)} = y^n`. For reference,
+    see [gst-2001]_.
+
+    .. [gst-2001]
+
+        S. Gottlieb, C.-W. Shu, E. Tadmor, *Strong Stability
+        Preserving High-Order Time Discretization Methods*, SIAM, Vol. 43,
+        pp. 89-112, 2001.
+        https://doi.org/10.1137/S003614450036757X
+    """
+
+    def __init__(self, component_id, state_filter_name=None, rhs_func_name=None):
+        super().__init__()
+
+        state_filter = state_filter_name
+        if state_filter is not None:
+            state_filter = var(f"<func>{state_filter_name}")
```

Comment:
What's this `state_filter` for? Everyone else seemed to have it, so I just added it here too.

Reply:
It's a hack to in-place-modify time integration state. Using this means that you have little hope of attaining the design order of your time stepper. But *cough* some *cough* spatial discretizations seem to need to have modal filters applied to the state in order to stay stable.

Reply:
I see! Why is that added as a post-processing step instead of as part of the `rhs_func`?